### ISSUE TITLE
`Style/SoleNestedConditional` should not trigger when variable is declared in condition.

### ISSUE THREAD

**Issue by antondalgren** [rubocop/rubocop#9603]
```ruby
❯ irb
irb(main):001:0> a = { a: 12 }
=> {:a=>12}
irb(main):002:1* if b = a[:a] && b
irb(main):003:1*   puts b
irb(main):004:0> end
=> nil
irb(main):005:0> a = { a: 12 }
=> {:a=>12}
irb(main):006:1* if b = a[:a]
irb(main):007:1*   puts b if b
irb(main):008:0> end
12
=> nil
irb(main):009:0> 
```

I tried to adress this issue my self, but I had a hard time understanding the AST, it's methods etc. 

--------

## Expected behavior

```ruby
a = { a: 12 }
if b = a[:a]
  puts b if b
end
```
## Actual behavior

```ruby
a = { a: 12 }
if b = a[:a] && b
  puts b
end
```

## Steps to reproduce the problem

Spec that fails:
```ruby
  it 'does not register an offense when using nested modifier on value declared in condition' do
    expect_no_offenses(<<~RUBY)
    if foo && bar = something_else
      do_something if bar
    end
    RUBY
  end
```
## RuboCop version

```
❯ rubocop -V
1.10.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 arm64-darwin20)
  - rubocop-minitest 0.10.3
  - rubocop-performance 1.10.1
```
